### PR TITLE
lib: location: wifi: nRF Cloud Wi-Fi over MQTT

### PR DESCRIPTION
**Pull request by jhirsi**
In addition to existing REST support, MQTT is now supported with NRF Cloud Wi-Fi positioning. 
NRF_CLOUD_LOCATION taken into use for nRF Cloud Wi-Fi. Thus, removing all nRF Cloud JSON codecs from location lib. 
Jira: NCSDK-15604

### CODE REVIEW

File: doc/nrf/libraries/modem/location.rst [hunk 1]
```diff
@@ -87,8 +88,7 @@ You can configure the required credentials for the location services using Kconf
 Wi-Fi chip
 ==========
 
-None of the supported DKs have a Wi-Fi chip. You can use external Wi-Fi chip, such as ESP8266, and connect it to the nRF9160 DK.
-You can see :ref:`location_sample` and its DTC overlay for some more information on ESP8266 integration.
+None of the supported DKs have an integrated Wi-Fi chip. You can use external Wi-Fi chip, such as nRF7002 EK, and connect it to the nRF9160 DK.
```
Comment (peknis) [inline doc/nrf/libraries/modem/location.rst:91]:
```suggestion
None of the supported DKs have an integrated Wi-Fi chip.
You can use an external Wi-Fi chip, such as nRF7002 EK, and connect it to the nRF9160 DK.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by trantanen — APPROVED** [review]
Good work!

File: doc/nrf/libraries/modem/location.rst (hunk 1 above)
Comment (jhirsi) [reply doc/nrf/libraries/modem/location.rst]: Changed.

**Review by peknis — APPROVED** [review]

**Review by hiltunent — APPROVED** [review]

**Comment by glarsennordic** [comment]
Thanks for this, @jhirsi !

**Pull request merged**